### PR TITLE
Add script to update gradle deps related files

### PR DESCRIPTION
**Pull request by rmoreliovlabs**
## Description
Added script in order to update project dependencies

## Motivation and Context
In the project we keep a file with hashes of all dependencies we reference in the gradle file - ./gradle/verification-metadata.xml.

If new deps are added or existing ones are updated, we need to re-generate this XML file using this command:

`./gradlew --no-daemon --refresh-dependencies --write-verification-metadata sha256 assemble`

More info [here](https://docs.gradle.org/6.5/userguide/dependency_verification.html#sec:troubleshooting-verificaton).

As an improvement, we may have a shell script in the project, which would simplify this process. That script can run that long gradle command to re-generate the verification-metadata.xml file.

The script could simplify life of external contributors, as well as internal devs

Also we may update our wiki to mention about this capability.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

**Review by Vovchyk — CHANGES_REQUESTED** [review]
`update-deps.sh`? also I'd put it into the `scripts` folder, what do you think?

seems this command adds new entries of new dependencies to the metadata file, but also preserves older ones. Do you know if this command has an option to remove older / obsolete entries from the xml file?

**Comment by rmoreliovlabs** [comment]
>

**Comment by rmoreliovlabs** [comment]
> `update-deps.sh`? also I'd put it into the `scripts` folder, what do you think?
> 
> seems this command adds new entries of new dependencies to the metadata file, but also preserves older ones. Do you know if this command has an option to remove older / obsolete entries from the xml file?

Addressed comment but also found this: https://docs.gradle.org/current/userguide/dependency_verification.html#sec:verification-metadata-hygiene.

Let me know what you think about it

File: scripts/update-verification-metadata [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+cd ../
```
Comment (Vovchyk) [inline scripts/update-verification-metadata:2]:
```suggestion

```

script is supposed to be executed in the project's root directory, so no need to `cd`

File: scripts/update-verification-metadata [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+rm gradle/verification-metadata.xml
+cp scripts/verification-metadata-template.xml gradle/verification-metadata.xml
+./gradlew --no-daemon --refresh-dependencies --write-verification-metadata sha256 assemble
```
Comment (Vovchyk) [inline scripts/update-verification-metadata:5]:
```suggestion
set -e

# remove the verification-metadata.xml file with old deps (if any), and replace it with the template
if [ -f ./gradle/verification-metadata.xml ]; then
  rm ./gradle/verification-metadata.xml
  cp ./scripts/verification-metadata-template.xml ./gradle/verification-metadata.xml
fi

# generate verification-metadata.xml
./gradlew --no-daemon --refresh-dependencies --write-verification-metadata sha256 assemble
```

better to fail ( via `set -e` ) and exit, if for some reason file cannot be removed or the template is missing

**Review by Vovchyk — CHANGES_REQUESTED** [review]
I'd also add the `.sh` extension to the script name to make it easily identifiable as a shell script

**Comment by Vovchyk** [comment]
@rmoreliovlabs , I see you marked it as resolved. What are your thoughts on this?

> I'd also add the .sh extension to the script name to make it easily identifiable as a shell script

Also please make sure the pipeline is green

**Comment by rmoreliovlabs** [comment]
> @rmoreliovlabs , I see you marked it as resolved. What are your thoughts on this?
> 
> > I'd also add the .sh extension to the script name to make it easily identifiable as a shell script
> 
> Also please make sure the pipeline is green

Sorry I forgot to push the changes

**Review by Vovchyk — APPROVED** [review]

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Review by Vovchyk — APPROVED** [review]

**Pull request merged**